pkg/controller/machine/machine_controller.go

```diff
@@ -245,6 +230,29 @@ func (r *ReconcileMachine) Reconcile(request reconcile.Request) (reconcile.Resul
 	return reconcile.Result{}, nil
 }
 
+// needUpdateFinalizers appends any missing finalizers to the machine
+// and returns true if the api server needs to be updated.
+func needUpdateFinalizers(m *clusterv1.Machine, cluster *clusterv1.Cluster) bool {
```

Comment:
How about `reconcileFinalizers`?

Reply:
The name of this function sounds like it's performing a check that doesn't have any side effects, but it does... +1 to something like what Vince suggested.